### PR TITLE
Replaced componentWillReceiveProps

### LINKED ISSUE #168: Remove componentWillReceiveProps
Hello,
create-react-app (v3) bundler is complaining about the use of componentWillReceiveProps and suggesting to rely on componentDidUpdate instead or at least rename it to UNSAFE_componentWillReceiveProps.

Thanks

### PR DESCRIPTION
**Pull request by juliaschafer**
This fixes #168

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,17 +22,24 @@ class Dropdown extends Component {
...
-      var selected = this.parseValue(newProps.value, newProps.options)
-      if (selected !== this.state.selected) {
-        this.setState({selected: selected})
+  componentDidUpdate (prevProps) {
+    if (this.props.value !== prevProps.value) {
+      if (this.props.value) {
+        var selected = this.parseValue(this.props.value, this.props.options)
+        if (selected !== this.state.selected) {
+          this.setState({ selected: selected })
+        }
```
Comment (iSunRise) [inline index.js]: I guess that these only 6 lines are changes it needs.

**Comment by iSunRise** [comment]
Hey @guischafer ,

I come here to fix "componentWillReceiveProps" warning as well and decided to look at your PR.
Unfortunately  there's a lot if changes not related to fix, mostly code style changes.
I'm not sure whether you did them intentionally or it was done by IDE automatically, but since that changes are preset I don't think this PR will be accepted.

It would be great if you undo the style changes and left the functionality changes only.
Thanks

**Comment by guischafer** [comment]
> Hey @guischafer ,
> 
> I come here to fix "componentWillReceiveProps" warning as well and decided to look at your PR.
> Unfortunately there's a lot if changes not related to fix, mostly code style changes.
> I'm not sure whether you did them intentionally or it was done by IDE automatically, but since that changes are preset I don't think this PR will be accepted.
> 
> It would be great if you undo the style changes and left the functionality changes only.
> Thanks

Sorry about that, indeed my IDE was configured to auto format on save with a style guide of another project. I will fix it when I arrive home from work.

**Comment by iambumblehead** [comment]
bump @iSunRise

**Comment by iSunRise** [comment]
Well, sorry, I'm not related to this project, you should contact @fraserxu

**Comment by iambumblehead** [comment]
This package seems abandoned

**Comment by fraserxu** [comment]
Sorry for the delay, change has been deployed to `react-dropdown@1.8.0`

**Pull request merged**